B_adding_pages.md

```diff
@@ -222,7 +222,15 @@ There are a couple of things to notice here. We pattern match against the params
 
 Within the body of the `show` action, we also pass a third argument into the render function, a key value pair where `:messenger` is the key, and the messenger variable is passed as the value.
 
-It's good to remember that the keys to the params [Dict](http://elixir-lang.org/docs/stable/elixir/Dict.html) will always be strings.
+Note that if the code in the body of the action needs access to the entire map of parameters, the `show` function can be defined this way instead:
```

Comment:
I have a question about this line. If we really just wanted all the params, we could just do this:
`def show(conn,  params) do`

What we are doing with the line that we do have:
`def show(conn, %{"messenger" => messenger} = params) do`

is to pattern match on the `params` map we do get passed to make sure it has a `"messenger"` key, and bind the `messenger` variable to the value of that key while still making the whole `params` map available as `params`.

Here's hoping that sentence makes sense outside my own head. :^)